matsim/src/main/java/org/matsim/core/router/NetworkRoutingInclAccessEgressModule.java

```diff
@@ -299,7 +288,9 @@ public String toString() {
 
 		if (toLink != fromLink) { // (a "true" route)
 
```

Comment:
So, if one constucts a scenario as I did in the test, this will be executed before PrepareForSim. Therefore in the 0th iteration the vehicle of a user is not known since it hasn't been added to the scenario in PrepareForSim. What do you think @kainagel ?

Reply:
Could you please let me know which test so I could try myself/see the problem?  Thanks ...

Reply:
https://github.com/matsim-org/matsim/pull/596/files#diff-0210a2a5d9f242ce5ed2105852b9c5fe
Sorry about the lengthy file. I meant to clean it up before I merge it into master